src/server/index.jsx

```diff
@@ -104,14 +104,20 @@ server
 
       const headHelmet = Helmet.renderStatic();
 
-      const styleTags = sheet.getStyleElement();
+      let inlineCss = '';
+      try {
+        // eslint-disable-next-line prefer-destructuring
+        inlineCss = renderToString(sheet.getStyleElement()).match(
+          /^<style (?:[^>]+)>([\S\s.]+)<\/style>$/,
```

Comment:
We should pull out the regex logic into a method `styleTagRegex` that is in a separate file, e.g. `src/server/styleRegex.js`. It should have unit tests that capture various examples that we would expect to capture, as well as examples that it shouldn't capture.

The reason we should do this is that the regex - in conjunction with the `getStyleElement` function provided by Styled Components - is fundamental to us having correct loading of styles on our article pages.  So having more detailed unit tests could give us confidence that this will not break.

We should also ensure that we have sufficient tests surrounding the `getStyleElement()` method as it is used in our application. To start with, it would be useful to split this into several lines, to make debugging simpler.

```
const styledComponentStyleElements = sheet.getStyleElement();
const allStyles = renderToString(styledComponentsStyleElements).match(styleRegex);
const inlineCss = allStyles[1]; // ((Explanation why second style tag is inlined))
```
Then, we could have error handling for the case where `allStyles` is empty. 

Splitting up the logic over multiple lines will enhance readability and maintainability.

Reply:
Thanks. Totally agreed, except that current PR only allows for one style tag, so I need to fundamentally change how it works as per https://github.com/BBC-News/simorgh/pull/921/files#r234940648.

Implemented in https://github.com/BBC-News/simorgh/pull/921/commits/691ebb6bbdf3637954d8809135926b29945bf09e.